src/main/java/Assistinator/Deadline.java

```diff
@@ -0,0 +1,35 @@
+package Assistinator;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    protected LocalDate by;
+
+    /**
+     * Initialise deadline task
+     * @param description task description
+     * @param by deadline
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = LocalDate.parse(by);
+    }
+
+    /**
+     * Convert task to formatted string
+     * @return formatted string
+     */
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
+    }
+
+    /**
+     * Generate string to print onto file
+     * @return String for file
+     */
+    public String toFileString() {
```

Comment:
Consider naming `toStorageString` instead of `toFileString` as could be confused for file path string